framework/include/vectorpostprocessors/VectorPostprocessorData.h

```diff
@@ -90,16 +128,32 @@ class VectorPostprocessorData : public Restartable
   const std::vector<std::pair<std::string, VectorPostprocessorState>> &
   vectors(const std::string & vpp_name) const;
 
+private:
+  // Forwad declaration
+  struct VectorPostprocessorVectors;
+
+public:
```

Comment:
Can we reorganize this a bit? You are introducing two new labels so we are switching back and forth between public/private multiple times in this header.